tests/queries/0_stateless/02269_create_table_with_collation.sql

```diff
@@ -0,0 +1,7 @@
+SET compatibility_ignore_collation_in_create_table=false;
+CREATE TABLE t_02267_collation (x varchar(255) COLLATE utf8_unicode_ci NOT NULL) ENGINE = Memory; --{serverError 49}
+
+SET compatibility_ignore_collation_in_create_table=true;
+CREATE TABLE t_02267_collation (x varchar(255) COLLATE utf8_unicode_ci NOT NULL) ENGINE = Memory;
+
```

Comment:
It's also good to add tests to check that it properly formatted with `EXPLAIN SYNTAX`

Reply:
Ok , I will add `EXPLAIN SYNTAX` to the test file.